src/Cbor.cpp

```diff
@@ -0,0 +1,458 @@
+// Copyright © 2017-2019 Trust Wallet.
+//
+// This file is part of Trust. The full Trust copyright notice, including
+// terms governing use, modification, and redistribution, is contained in the
+// file LICENSE at the root of the source code distribution tree.
+
+#include "Cbor.h"
+#include "HexCoding.h"
+
+#include <sstream>
+
+namespace TW::Cbor {
+
+using namespace std;
+
+
+Encode::Encode(const TW::Data& rawData)
+{
+    // check validity, may throw
+    Decode check(rawData);
+    if (!check.isValid()) { throw invalid_argument("Invalid CBOR data"); }
+    data = rawData;
+}
+
+Encode Encode::uint(uint64_t value) {
+    Encode e;
+    e.appendValue(Decode::MT_uint, value);
+    return e;
+}
```

Comment:
This pattern looks quite repetitive, maybe we can pass `Encoder` reference in? so no need to return it